presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneTableScanColumns.java

```diff
@@ -72,11 +69,11 @@ public Pattern getPattern()
                                 child.getId(),
                                 child.getTable(),
                                 newOutputs,
-                                newOutputs.stream()
-                                        .collect(Collectors.toMap(Function.identity(), e -> child.getAssignments().get(e))),
+                                filterKeys(child.getAssignments(), newOutputs::contains),
```

Comment:
The only purpose of newOutputs is to preserve the previous order of outputs to not change the node unnecessarily.  Using dependencies.get()::contains in filterKeys would in my opinion be more direct and would scale properly with the number of columns.